src/Mvc/src/Microsoft.AspNetCore.Mvc.Core/Authorization/AuthorizeFilter.cs

```diff
@@ -136,38 +134,30 @@ private async Task<AuthorizationPolicy> GetEffectivePolicyAsync(AuthorizationFil
             var effectivePolicy = await ComputePolicyAsync();
             var canCache = PolicyProvider == null;
 
-            if (_mvcOptions == null) 
+            if (!context.IsEffectivePolicy(this))
```

Comment:
Not sure why this check was behind the `AllowCombiningAuthorizeFilters` option. That required some seemingly-unrelated test changes.

In any case, should this be above the `ComputePolicyAsync()` call or in `OnAuthorizationAsync(...)`, prior to calling `GetEffectivePolicyAsync(...)`. The extra `async` state machines for this sync call seem wasteful.

Reply:
Yeah, that seems wonky. There's an issue tracking cleaning this type up in 3.0: https://github.com/aspnet/AspNetCore/issues/4862

Reply:
👍

Reply:
Yeah I would imagine so.

Reply:
@rynowak should I reorder things in this PR or just add a comment in #4862?